src/main/java/app/model/agents/AgentImp.java

```diff
@@ -47,13 +50,14 @@ public AgentImp(Vector position, Vector direction, double radius, Team team)
         heard = new ArrayList<>();
     }
 
-    public AgentImp(Vector position, Vector direction, double radius, Team team, Vector tgtDirection)
+    public AgentImp(Vector position, Vector direction, double radius, Team team, Vector tgtDirection, Furniture target)
```

Comment:
I think this may cause some issues with the sub classes, as every one will need an amendment as well for a new target.  I think a cleaner implementation will be possible if we instead "see" the target using the visible rays.  This would link closely with the work that @mahshidghaffari is doing for a similar problem